fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -72,24 +98,31 @@ def load_data(self, X, y, event=None, **kwargs):
 
     def gamma(self, predictor):
         """Calculate the degree to which constraints are currently violated by the predictor."""
-        pred = predictor(self.X)
+        utility_diff = self.utility[_POSITIVE_UTILITY] - self.utility[_NEGATIVE_UTILITY]
+        pred = utility_diff * predictor(self.X) + self.utility[_NEGATIVE_UTILITY]
         self.tags[_PREDICTION] = pred
         expect_event = self.tags.groupby(_EVENT).mean()
         expect_group_event = self.tags.groupby(
             [_EVENT, _GROUP_ID]).mean()
-        expect_group_event[_DIFF] = expect_group_event[_PREDICTION] - expect_event[_PREDICTION]
-        g_unsigned = expect_group_event[_DIFF]
-        g_signed = pd.concat([g_unsigned, -g_unsigned],
+        expect_group_event[_UPPER_BOUND_DIFF] = self.ratio * \
```

Comment:
Can we get these on fewer lines? I think you have 99 characters available.

Reply:
I didn't realise. I'll refactor this.

Reply:
If you run `flake8` from the repo root directory you should get a list of all issues like lines that are too long. I suggest ignoring such things by default and running flake8 before committing :-)